Oneshot mode

ana-cc opened the pull request:
Adds a oneshot mode to onionperf. This uses a new tgen model to attempt downloading a 5M file. Once a download has finished, onionperf exits gracefully.

Commit:
Adds a tgen model for transferring 5m files in oneshot mode

Commit:
Adds support for oneshot mode cli argument

Commit:
Adds flow control logic to deal with oneshot mode

irl commented on onionperf/onionperf:
```diff
@@ -163,12 +163,12 @@ def main():
         default=8080)
 
-    measure_parser.add_argument('-i', '--no-inet',
+    measure_parser.add_argument('-i', '--onion-only',
```
Can we make this -o?

irl commented on onionperf/onionperf:
```diff
@@ -163,12 +163,12 @@ def main():
         default=True)
 
-    measure_parser.add_argument('-o', '--no-onion',
+    measure_parser.add_argument('-o', '--inet-only',
```
And this one -i?

irl commented on onionperf/measurement.py:
```diff
@@ -300,6 +312,8 @@ def __start_tgen(self, name, tgen_port, socks_port=None, server_urls=None):
             model.ListenModel(tgen_port="{0}".format(tgen_port)).dump_to_file(tgen_confpath)
             logging.info("TGen server running at 0.0.0.0:{0}".format(tgen_port))
+        elif self.oneshot:
+            model.OneshotModel(tgen_port="{0}".format(tgen_port), tgen_servers=server_urls, socksproxy="127.0.0.1:{0}".format(socks_port)).dump_to_file(tgen_confpath)
```
If these arguments are the same, can we instead assign the model to a variable and then call that?

ana-cc committed:
Update onionperf

ana-cc committed:
Assigns tgen model to variable, formats arguments in a dictionary

Commit:
Removes non-deterministic loop exit condition

Commit:
Ensures OP exits oneshot mode when multiple downloads happen before checking logfile

Commit:
Ensures tor is not run as daemon by overriding option in base config string

irl commented on onionperf/measurement.py:
```diff
@@ -68,9 +68,8 @@ def watchdog_thread_task(cmd, cwd, writable, done_ev, send_stdin, ready_search_s
         while subp.poll() is None and done_ev.is_set() is False:
             try:
                 # collect lines until the queue is empty for a full second
```
This comment is not valid anymore.

ana-cc replied:
Have removed comment in 852e90e

ana-cc committed:
Removes unnecessary comment from code

ana-cc committed:
Ensures tgen client models are not generated for tgen server

This change ensures Tgen client models, which require values for server_urls, are not generated when starting a tgen server. This flow logic was lost in previous commit on this branch 9c4575a

irl commented:
Rebased and merged.
